### ISSUE TITLE
QuotaExceededError

### ISSUE THREAD

**Issue by Yaffle** [jakearchibald/idb-keyval#26]
fill your storage device to have no space, then run next code:
```
var value = 'a'.repeat(50 * 1024 * 1024);
idbKeyval.set('hello', value).finally(x => console.log(x));
```
`QuotaExceededError` will cause `transaction.onabort`, but no `transaction.onerror` (at least in Chrome 66) as there is no `onabort` listener, the promise will never be resolved/rejected, seems.

P.S. I was trying to find a good example of IndexedDB usage and found your lib, I don't use it yet.

**Comment by jakearchibald** [comment]
Good catch.

You may be interested in https://github.com/jakearchibald/idb, which already treats aborts as rejections.

**Comment by jakearchibald** [comment]
This is fixed in 2.5.1 & 3.0.0.

https://github.com/jakearchibald/idb-keyval/commit/90e004c83b867722c649f66d8d16dd85c4a2d061#diff-32d9810b60fe9355ae31121d8cc181e2R23

Thanks for reporting it!

**Comment by Yaffle** [comment]
Thanks.
By the way, `transaction.onerror` is redundant, as `abort` event follows `error` events (unless someone does `event.preventDefault()`)

**Comment by jakearchibald** [comment]
Good point. That'll save a couple of bytes.

**Comment by jakearchibald** [comment]
This is fixed in 4.0.0
